services/meminator-ruby/src/app.rb

```diff
@@ -0,0 +1,77 @@
+require 'sinatra/base'
+require 'open-uri'
+require 'tempfile'
+
+class MeminatorApp < Sinatra::Application
+  set :bind, '0.0.0.0'
+  set :port, '10117'
+  set :show_exceptions, false
+
+
+  IMAGE_MAX_WIDTH_PX = 1000
+  IMAGE_MAX_HEIGHT_PX = 1000
+
+  post '/applyPhraseToPicture' do
+    request.body.rewind
+    work = JSON.parse(request.body.read)
+    logger.info "Received request: #{work}"
+
+    phrase = work.fetch("phrase") { return [400, "Missing 'phrase' key in JSON request body."] }
+    image_url = work.fetch("imageUrl") { return [400, "Missing 'imageUrl' key in JSON request body."] }
+
+    begin
+      image_url = URI.parse(image_url)
+      [URI::HTTP, URI::HTTPS].include?(image_url.class) or raise URI::InvalidURIError
+    rescue URI::InvalidURIError
+      return [400, "Invalid 'imageUrl' value in JSON request body."]
+    end
+
+    file_extension = File.extname(image_url.path)
+    downloaded_image = Tempfile.new(["downloaded_image", file_extension])
+    downloaded_image.binmode
+    downloaded_image.write(URI.open(image_url).read)
```

Comment:
Kinda curious if instrumentation picks up on this HTTP request. I predict we'll need to use a proper HTTP client instead of `URI.open()`.